driver/pace/driver/run.py

```diff
@@ -459,47 +549,55 @@ def __init__(
             comm: communication object behaving like mpi4py.Comm
         """
         self.config = config
-        communicator = pace.util.CubedSphereCommunicator.from_layout(
-            comm=comm, layout=self.config.layout
-        )
-        quantity_factory, stencil_factory = _setup_factories(
-            config=config, communicator=communicator
-        )
+        self.comm = comm
+        self.performance_config = self.config.performance_config
+        self.timestep_timer = self.performance_config.timestep_timer
+        self.timer = self.performance_config.total_timer
+        self.times_per_step = self.performance_config.times_per_step
```

Comment:
why are we propagating these variables out here?

Reply:
Moved them to class attributes instead of properties.